### PR TITLE
Upgrade all dependencies

### PR DESCRIPTION
**Pull request by DemiMarie-parity**
This upgrades all dependencies to the latest version.

### CODE REVIEW

File: srml/contracts/src/tests.rs [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ fn call_removed_contract() {
 				"contract has been evicted"
 			);
 
- 			// Subsequent contract calls should also fail.
+		// Subsequent contract calls should also fail.
```
Comment (pepyakin) [inline srml/contracts/src/tests.rs]: Not sure if this is needed.

File: core/state-machine/src/lib.rs [hunk 2]
```diff
@@ -1241,7 +1241,7 @@ mod tests {
 			remote_proof.clone(),
 			&[0xff]
 		).is_ok();
- 		// check that results are correct
+	// check that results are correct
```
Comment (pepyakin) [inline core/state-machine/src/lib.rs]: Are these intended changes?

File: core/state-machine/src/changes_trie/storage.rs [hunk 3]
```diff
@@ -67,7 +67,7 @@ impl<H: Hasher, Number: BlockNumber> InMemoryStorage<H, Number> {
 	pub fn with_proof(proof: Vec<Vec<u8>>) -> Self {
 		use hash_db::HashDB;
 
- 		let mut proof_db = MemoryDB::<H>::default();
+	let mut proof_db = MemoryDB::<H>::default();
```
Comment (pepyakin) [inline core/state-machine/src/changes_trie/storage.rs]: Wrong indent

File: core/rpc/src/author/tests.rs [hunk 4]
```diff
@@ -147,7 +147,7 @@ fn should_return_pending_extrinsics() {
 	};
 	let ex = uxt(AccountKeyring::Alice, 0);
 	AuthorApi::submit_extrinsic(&p, ex.encode().into()).unwrap();
- 	assert_matches!(
+assert_matches!(
```
Comment (pepyakin) [inline core/rpc/src/author/tests.rs]: wrong indent

File: core/rpc/src/author/tests.rs [hunk 5]
```diff
@@ -180,7 +180,7 @@ fn should_remove_extrinsics() {
 		hash::ExtrinsicOrHash::Extrinsic(ex1.encode().into()),
 	]).unwrap();
 
- 	assert_eq!(removed.len(), 3);
+assert_eq!(removed.len(), 3);
```
Comment (pepyakin) [inline core/rpc/src/author/tests.rs]: wrong indent

File: core/network/src/protocol.rs [hunk 6]
```diff
@@ -1434,7 +1434,7 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 				None
 			},
 		};
- 		self.send_message(
+	self.send_message(
```
Comment (pepyakin) [inline core/network/src/protocol.rs]: wrong indent

**Comment by pepyakin** [comment]
Also, since this PR does change logic, unmarking it as insubstantial. 

There is some code changed in support, cc/ @thiolliere / @kianenigma

**Comment by thiolliere** [comment]
putting quote and syn in individual PR might be better. generally speaking splitting PR is a good practice - having 1 insubstantial and 2 easy PR is better than one somewhat difficult to read one.

**Comment by bkchr** [comment]
> putting quote and syn in individual PR might be better. generally speaking splitting PR is a good practice - having 1 insubstantial and 2 easy PR is better than one somewhat difficult to read one.

Yeah. Updating dependencies should not involve updating to breaking releases. Breaking releases should be handled in different prs.

**Comment by kianenigma** [comment]
(I am offline for another two days and actually don't have a laptop. and loading the diff page crashes my phone browser, so no can review sadly)

Highly agreeing with @thiolliere: there should be a PR that updates only those depa that do not require any code change to pass the ci. Breaking changes should be handled individually)

**Comment by DemiMarie-parity** [comment]
@kianenigma That is a good idea, thanks!  Some of the changes actually depend on an unreleased version of `impl-serde`, and I may need to switch from `once_cell` to `lazy_static` to get the proc macros working.

**Comment by DemiMarie-parity** [comment]
Closing this PR so that I can split it into smaller ones.

**Pull request closed without merging**